website: DynamoDB now supports fine-grain IAM policies (#29227)
hashicorp/terraform: website/docs/language/settings/backends/s3.html.md

}
```

> DynamoDB doesn't assign a separate resource ARN to each key in a table, but

Reviewer: ```suggestion
DynamoDB does not assign a separate resource ARN to each key in a table, but
```

you can write more precise policies for a DynamoDB table
[using an IAM `Condition` element](https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/specifying-conditions.html).
For example, you can use the `dynamodb:LeadingKeys` condition key to match
on the partition key values that the S3 backend will use:

```json
{
  "Version": "2012-10-17",
  "Statement": [
    {
      "Effect": "Allow",
      "Action": [
        "dynamodb:DeleteItem",
        "dynamodb:GetItem",
        "dynamodb:PutItem"
      ],
      "Resource": "arn:aws:dynamodb:us-east-1:12341234:table/TableName",
      "Condition": {
        "ForAllValues:StringEquals": {
          "dynamodb:LeadingKeys": [
            "TableName/myapp/production/tfstate",
            "TableName/myapp/production/tfstate-md5"
          ]
        }
      }
    }
  ]
}
```

> In the above, `TableName` represents the table name given in the

Reviewer: ```suggestion
In the example above, `TableName` represents the table name given in the
```

`dynamodb_table` argument in the backend configuration, and
`myapp/production/tfstate` represents the path to the particular S3 object
for this environment's state. DynamoDB ARNs are regional and account-specific,

> unlike S3 bucket ARNs, so be sure to also specify the correct region and

Reviewer: ```suggestion
unlike S3 bucket ARNs, so you should also specify the correct region and
```
